Sources/ForageSDK/Foundation/Network/Collector.swift

```diff
@@ -7,37 +7,166 @@
 
 import Foundation
 import VGSCollectSDK
+import BasisTheoryElements
 
-internal class Collector {
-    /**
-     VGS VaultId
-     */
-    private enum VaultId: String {
-        case sandbox = "tntagcot4b1"
-        case cert = "tntpnht7psv"
-        case prod = "tntbcrncmgi"
-        case staging = "tnteykuh975"
-        case dev = "tntlqkidhc6"
-    }
+public protocol VaultCollector {
+    func setCustomHeaders(headers: [String: String])
+    func sendData(
+        path: String,
+        extraData: [String: Any],
+        completion: @escaping (VGSResponse) -> Void)
+    
+}
+
+struct VGSCollectConfig {
+    let id: String
+    let environment: VGSCollectSDK.Environment
+}
+
+struct BasisTheoryConfig {
+    let publicKey: String
+    let proxyKey: String
+}
+
+// Wrapper class for VGSCollect
+class VGSCollectWrapper: VaultCollector {
+    
+    public let vgsCollect: VGSCollect
     
-    public static func CreateVGS() -> VGSCollect {
-        return VGSCollect(id: vaultID(ForageSDK.shared.environment).rawValue, environment: environmentVGS(ForageSDK.shared.environment))
+    init(config: VGSCollectConfig) {
+        self.vgsCollect = VGSCollect(id: config.id, environment: config.environment)
```

Comment:
Important callout that I had to adjust before, we need to always be using a new VGS instance for each input field. Otherwise, two input fields on the same screen will step on each other's toes.

Reply:
i don't think this would conflict with that?

Reply:
Ah yes had this confused with the collector factory function below. You're good.